components/layout/block.rs

```diff
@@ -1390,8 +1390,20 @@ impl BlockFlow {
             // and its inline-size is our content inline-size.
             let kid_mode = flow::base(kid).writing_mode;
             {
+                // Don't assign positions to children unless they're going to be reflowed.
+                // Otherwise, the position we assign might be incorrect and never fixed up. (Issue
+                // #13704.)
+                //
+                // For instance, floats have their true inline position calculated in
+                // `assign_block_size()`, which won't do anything unless `REFLOW` is set. So, if a
+                // float child does not have `REFLOW` set, we must be careful to avoid touching its
+                // inline position, as no logic will run afterward to set its true value.
                 let kid_base = flow::mut_base(kid);
-                if kid_base.flags.contains(INLINE_POSITION_IS_STATIC) {
+                if kid_base.flags.contains(INLINE_POSITION_IS_STATIC) &&
+                        ((kid_base.flags.contains(IS_ABSOLUTELY_POSITIONED) &&
+                          kid_base.restyle_damage.contains(REFLOW_OUT_OF_FLOW)) ||
+                         (!kid_base.flags.contains(IS_ABSOLUTELY_POSITIONED) &&
+                          kid_base.restyle_damage.contains(REFLOW))) {
```

Comment:
Ugh, parenthesis counting. Maybe it would be better to break it up into a couple let bindings?

```rust
let inline_position_static = kid_base.flags.contains(INLINE_POSITION_IS_STATIC);
let absolute_and_reflow_out_of_flow = kid_base.flags.contains(IS_ABSOLUTELY_POSITIONED) && kid_base.restyle_damage.contains(REFLOW_OUT_OF_FLOW);
let not_absolute_and_reflow = !kid_base.flags.contains(IS_ABSOLUTELY_POSITIONED) && kid_base.restyle_damage.contains(REFLOW);

if is_static && (absolute_and_reflow_out_of_flow || not_absolute_and_reflow) {
```